pkgs/by-name/we/wesnoth/package.nix

```
@@ -73,10 +79,33 @@ stdenv.mkDerivation (finalAttrs: {
 
   cmakeFlags = [
     "-DENABLE_SYSTEM_LUA=ON"
+  ]
+  ++ lib.optional enableDevel [
+    "-DBINARY_SUFFIX=-devel"
   ];
 
+  patchPhase = lib.optionalString enableDevel ''
```

Use postPatch instead of overriding the whole patchPhase:

```suggestion
  postPatch = lib.optionalString enableDevel ''
```